ledger/tracker.go

```diff
@@ -693,5 +697,16 @@ func (tr *trackerRegistry) initializeTrackerCaches(l ledgerForTracker) (err erro
 		err = blockRetrievalError
 	}
 	return
+}
+
+func (tr *trackerRegistry) blockTimestamp(rnd basics.Round) (int64, error) {
+	tr.mu.RLock()
+	defer tr.mu.RUnlock()
+	return tr.txtail.blockTimestamp(rnd, tr.dbRound)
+}
 
+func (tr *trackerRegistry) blockSeed(rnd basics.Round) ([]byte, error) {
+	tr.mu.RLock()
+	defer tr.mu.RUnlock()
+	return tr.txtail.blockSeed(rnd, tr.dbRound)
```

Comment:
Could he export the blockHeaderData struct instead, so that adding things to it (say we decide to hold onto the block hash, and/ or the new sha256 blockhash) doesnt require new methods on the trackerRegistry and on the txtail tracker. Instead, we just have one blockHdr() type method that might return a a bit more later.

Maybe it could be called `BlockHdrTail` or something like that to convey that it's the parts of the block header we retain for the taill.

I'm actually slightly thinking we should just retain the whole block header.  How big is it? Even if it's 1k (I'm sure it's smaller than that), it would only me 1mb of memory total, and we'd get rid of a few extra types.

Reply:
Yes, the main reason why did I go with separate fields is the potential large memory footprint. The struct is basically there - `txTailBlockHeaderData`, I guess I'll make it exported and available for ledger clients.

Reply:
But we only hold 1001 of them, so even if we hold the entire `bookkeeping.BlockHeader` that seems like less than 1mb of memory.  It's not like the per txn stuff like leases, where we have to be careful because 1000 blocks could be a million txns. I'm not pushing hard for it, but it seems easy and cheap.